### PR TITLE
bleach: Loosen attribute dict type

### PR DESCRIPTION
**Pull request by srittau**
Closes: #5650

### CODE REVIEW

File: stubs/bleach/bleach/sanitizer.pyi [hunk 1]
```diff
@@ -38,7 +38,9 @@ class Cleaner(object):
     def clean(self, text: Text) -> Text: ...
 
 _AttributeFilter = Callable[[Text, Text, Text], bool]
-_AttributeDict = Dict[Text, Union[Container[Text], _AttributeFilter]]
+_AttributeDict = Union[
+    Dict[Text, Union[Container[Text], _AttributeFilter]], Dict[Text, Container[Text]], Dict[Text, _AttributeFilter]
```
Comment (Akuli) [inline stubs/bleach/bleach/sanitizer.pyi:42]: Dict is invariant in value type, so this doesn't solve the problem: `dict[str, list[str]]` is not compatible with `dict[str, Union[...]]`. Does it really need a dict or can we get away with `Mapping` or a protocol?

Comment (srittau) [reply stubs/bleach/bleach/sanitizer.pyi:42]: But it is compatible with `dict[str, Container[str]]`, unless I'm missing something.

bleach uses an isinstance check, so it requires a real `dict`.

Comment (Akuli) [reply stubs/bleach/bleach/sanitizer.pyi:42]: `stdlib/builtins.pyi` uses `_VT` as the value type. It's defined as `TypeVar("_VT")`, without explicitly saying it's covariant or contravariant, so it is invariant.

Comment (Akuli) [reply stubs/bleach/bleach/sanitizer.pyi:42]: Here's a short example showing this:

```
(env) akuli@akuli-desktop:~/typeshed$ mypy --custom-typeshed-dir=. foo.py 
foo.py:5: error: Incompatible types in assignment (expression has type "Dict[str, List[str]]", variable has type "Dict[str, Container[str]]")
foo.py:5: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
foo.py:5: note: Consider using "Mapping" instead, which is covariant in the value type
Found 1 error in 1 file (checked 1 source file)

(env) akuli@akuli-desktop:~/typeshed$ cat foo.py
from collections.abc import Container

def main() -> None:
    a: dict[str, list[str]] = {"x": ["y"]}
    b: dict[str, Container[str]] = a
```

Comment (srittau) [reply stubs/bleach/bleach/sanitizer.pyi:42]: I guess we should just use `List` instead of `Container` then.

Comment (Akuli) [reply stubs/bleach/bleach/sanitizer.pyi:42]: Then it won't work for tuples, even though they would pass the `isinstance(foo, dict)` check. Maybe just `Any`?

Comment (srittau) [reply stubs/bleach/bleach/sanitizer.pyi]: When someone is complaining about using tuples, we can always add that case later. Or we can add it now if you think it's likely someone is using tuples.

**Review by Akuli — APPROVED** [review]
Feel free to merge. This looks reasonable, although I would have used `Any` in the spirit of "avoid false positives".

**Pull request merged**